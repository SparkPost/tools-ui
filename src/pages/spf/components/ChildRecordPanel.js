import React from 'react';
import classNames from 'classnames';

const ChildRecordPanel = (props) => {
  const {childrenCollapsed, type, value, record, children, toggle} = props;

  // TODO see if we can anchor link to panel with error
  // TODO add child type 'error'

  // Can't check children as they are not passed down if collapsed
  const isExpandable = type === 'redirect' || type === 'include';

  const panelClasses = classNames('panel spf-tree__child', {
    [`spf-tree__child--${type}`]: type,
    'can-expand': isExpandable
  });

  const iconClasses = classNames('spf-tree__chevron fa fa-chevron-down', {
    'is-open': childrenCollapsed
  });

  const labelClasses = classNames('spf-tree__code', {
    [`spf-tree__code--${type}`]: type,
    'spf-tree__code--label': record // if record, use label styles
  });

  return (
    <div className='spf-tree__childWrapper'>
      <div className={panelClasses}>
        <div className='panel__body' onClick={ toggle }>

          {isExpandable && <i className={iconClasses} />}
          <code className={labelClasses}>{ type }:{ value }</code>
          { record && <code className='spf-tree__code'>{ record }</code>}

        </div>
        <div className='spf-tree__accent' />
      </div>
      {children && <div className='spf-tree__children'>{ children }</div>}
    </div>
  );
};

ChildRecordPanel.defaultProps = {
  childrenCollapsed: true,
  type: '',
  value: '',
  toggle: null
};

ChildRecordPanel.propTypes = {
  childrenCollapsed: React.PropTypes.bool,
  type: React.PropTypes.string,
  value: React.PropTypes.string,
  record: React.PropTypes.string,
  children: React.PropTypes.array,
  toggle: React.PropTypes.func
};


export default ChildRecordPanel;
