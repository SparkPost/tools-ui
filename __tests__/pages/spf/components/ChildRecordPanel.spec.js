import React from 'react';
import renderer from 'react-test-renderer';
import ChildRecordPanel from 'pages/spf/components/ChildRecordPanel';

describe('ChildRecordPanel Snapshots', () => {
  let type
    , value
    , record
    , collapsed
    , childrenCollapsed;

  beforeEach(() => {
    type = 'include';
    value = '_spf.erlock.homes';
    record = 'mx';
    collapsed = false;
    childrenCollapsed = true;
  })

  test('baseline snapshot', () => {
    const panel = <ChildRecordPanel type={ type } value={ value } record={ record } collapsed={ false } childrenCollapsed={ true }></ChildRecordPanel>;
    expect(renderer.create(panel)).toMatchSnapshot();
  });

  test('should not show record if missing', () => {
    const panel = <ChildRecordPanel type={ type } value={ value }  collapsed={ false } childrenCollapsed={ true }></ChildRecordPanel>;
    expect(renderer.create(panel)).toMatchSnapshot();
  });

  test('should be hidden if collapsed is true', () => {
    const panel = <ChildRecordPanel type={ type } value={ value } record={ record } collapsed={ true } childrenCollapsed={ true }></ChildRecordPanel>;
    expect(renderer.create(panel)).toMatchSnapshot();
  });

  test('should change toggle icon if childrenCollapsed is false', () => {
    const panel = <ChildRecordPanel type={ type } value={ value } record={ record } collapsed={ false } childrenCollapsed={ false }></ChildRecordPanel>;
    expect(renderer.create(panel)).toMatchSnapshot();
  });
});
