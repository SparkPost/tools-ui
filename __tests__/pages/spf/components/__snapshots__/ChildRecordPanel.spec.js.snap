exports[`ChildRecordPanel Snapshots baseline snapshot 1`] = `
<div
  className="spf-tree__childWrapper">
  <div
    className="panel spf-tree__child spf-tree__child--include can-expand">
    <div
      className="panel__body"
      onClick={[Function]}>
      <i
        className="spf-tree__chevron fa fa-chevron-down is-open" />
      <code
        className="spf-tree__code spf-tree__code--include spf-tree__code--label">
        include
        :
        _spf.erlock.homes
      </code>
      <code
        className="spf-tree__code">
        mx
      </code>
    </div>
    <div
      className="spf-tree__accent" />
  </div>
</div>
`;

exports[`ChildRecordPanel Snapshots should be hidden if collapsed is true 1`] = `
<div
  className="spf-tree__childWrapper">
  <div
    className="panel spf-tree__child spf-tree__child--include can-expand">
    <div
      className="panel__body"
      onClick={[Function]}>
      <i
        className="spf-tree__chevron fa fa-chevron-down is-open" />
      <code
        className="spf-tree__code spf-tree__code--include spf-tree__code--label">
        include
        :
        _spf.erlock.homes
      </code>
      <code
        className="spf-tree__code">
        mx
      </code>
    </div>
    <div
      className="spf-tree__accent" />
  </div>
</div>
`;

exports[`ChildRecordPanel Snapshots should change toggle icon if childrenCollapsed is false 1`] = `
<div
  className="spf-tree__childWrapper">
  <div
    className="panel spf-tree__child spf-tree__child--include can-expand">
    <div
      className="panel__body"
      onClick={[Function]}>
      <i
        className="spf-tree__chevron fa fa-chevron-down" />
      <code
        className="spf-tree__code spf-tree__code--include spf-tree__code--label">
        include
        :
        _spf.erlock.homes
      </code>
      <code
        className="spf-tree__code">
        mx
      </code>
    </div>
    <div
      className="spf-tree__accent" />
  </div>
</div>
`;

exports[`ChildRecordPanel Snapshots should not show record if missing 1`] = `
<div
  className="spf-tree__childWrapper">
  <div
    className="panel spf-tree__child spf-tree__child--include can-expand">
    <div
      className="panel__body"
      onClick={[Function]}>
      <i
        className="spf-tree__chevron fa fa-chevron-down is-open" />
      <code
        className="spf-tree__code spf-tree__code--include">
        include
        :
        _spf.erlock.homes
      </code>
    </div>
    <div
      className="spf-tree__accent" />
  </div>
</div>
`;
